crates/compilers/src/compilers/resolc/compiler.rs

```diff
@@ -0,0 +1,713 @@
+            }
+            Err(_) => Ok(ResolcCompilerOutput::default()),
+        }
+    }
+
+    pub fn compile_output<T: Serialize>(&self, input: &ResolcInput) -> Result<Vec<u8>> {
+        let mut cmd = self.configure_cmd();
+        let mut child = cmd.spawn().map_err(|err| SolcError::io(err, &self.resolc))?;
+
+        let stdin = child.stdin.as_mut().unwrap();
+        serde_json::to_writer(stdin, input)?;
+
+        let output = child.wait_with_output().map_err(|err| SolcError::io(err, &self.resolc))?;
+
+        compile_output(output)
+    }
+
+    fn configure_cmd(&self) -> Command {
+        let mut cmd = Command::new(&self.resolc);
+        cmd.stdin(Stdio::piped()).stderr(Stdio::piped()).stdout(Stdio::piped());
+        cmd.args(&self.extra_args);
+        cmd.arg("--standard-json");
+        cmd
+    }
+}
+
+#[cfg(feature = "async")]
+fn compiler_blocking_install(
+    compiler_path: PathBuf,
+    lock_path: PathBuf,
+    download_url: &str,
+    label: &str,
+) -> Result<PathBuf> {
+    use foundry_compilers_core::utils::RuntimeOrHandle;
+    trace!("blocking installing {label}");
+    RuntimeOrHandle::new().block_on(async {
+        let client = reqwest::Client::new();
+        let response = client
+            .get(download_url)
+            .send()
+            .await
+            .map_err(|e| SolcError::msg(format!("Failed to download {label} file: {e}")))?;
+
+        if response.status().is_success() {
+            let content = response
+                .bytes()
+                .await
+                .map_err(|e| SolcError::msg(format!("failed to download {label} file: {e}")))?;
+            trace!("downloaded {label}");
+
+            trace!("try to get lock for {label}");
+            let _lock = try_lock_file(lock_path)?;
+            trace!("got lock for {label}");
+
+            if !compiler_path.exists() {
+                trace!("creating binary for {label}");
+                let mut output_file = File::create(&compiler_path).map_err(|e| {
+                    SolcError::msg(format!("Failed to create output {label} file: {e}"))
+                })?;
+
+                output_file.write_all(&content).map_err(|e| {
+                    SolcError::msg(format!("Failed to write the downloaded {label} file: {e}"))
+                })?;
+
+                set_permissions(&compiler_path, PermissionsExt::from_mode(0o755)).map_err(|e| {
+                    SolcError::msg(format!("Failed to set {label} permissions: {e}"))
+                })?;
+            } else {
+                trace!("found binary for {label}");
+            }
+        } else {
+            return Err(SolcError::msg(format!(
+                "Failed to download {label} file: status code {}",
+                response.status()
+            )));
+        }
+        trace!("{label} installation completed");
+        Ok(compiler_path)
+    })
+}
+
+#[cfg(feature = "async")]
```

Comment:
why it's marked as feature async if the api used is non async? shouldn't it be possible to use tokio API?